tools/strobelight/cli_function_profiler.py

```diff
@@ -0,0 +1,307 @@
+import functools
+import logging
+import os
+import re
+import subprocess
+import time
+from threading import Lock
+from typing import Any, List, Optional, Sequence
+
+# mypy: disallow-untyped-defs
+
+__all__ = [
+    "strobelight",
+    "StrobelightCLIProfilerError",
+    "StrobelightCLIFunctionProfiler",
+]
+
+logger = logging.getLogger("strobelight_function_profiler")
+
+console_handler = logging.StreamHandler()
+formatter = logging.Formatter(
+    "%(name)s, line %(lineno)d, %(asctime)s, %(levelname)s: %(message)s"
+)
+console_handler.setFormatter(formatter)
+
+logger.addHandler(console_handler)
+logger.setLevel(logging.INFO)
+logger.propagate = False
+
+
+class StrobelightCLIProfilerError(Exception):
+    """
+    Raised when an error happens during strobelight profiling
+    """
+
+
+def _pid_namespace_link(pid: Optional[int] = None) -> str:
+    PID_NAMESPACE_PATH = "/proc/{}/ns/pid"
+    """Returns the link to the process's namespace, example: pid:[4026531836]"""
+    pid = pid or os.getpid()
+    return os.readlink(PID_NAMESPACE_PATH.format(pid))
+
+
+def _pid_namespace(pid: Optional[int] = None) -> int:
+    """Returns the process's namespace id"""
+    pid = pid or os.getpid()
+    link = _pid_namespace_link(pid)
+    return int(link[link.find("[") + 1 : -1])
+
+
+def _command_to_string(command: Sequence[str]) -> str:
+    return " ".join(command)
+
+
+class StrobelightCLIFunctionProfiler:
+    """
+    Note: this is a meta only tool.
```

Comment:
capital 'M'eta.

Reply:
ah i missed this will address it innext one